static/src/javascripts/projects/facia/modules/onwards/weather.js

```diff
@@ -0,0 +1,212 @@
+// @flow
+/**
+    "WEATHER"
+
+    Whether the weather be fine,
+    Or whether the weather be not,
+    Whether the weather be cold,
+    Or whether the weather be hot,
+    We'll weather the weather
+    Whatever the weather,
+    Whether we like it or not!
+
+    Author: Anonymous British
+ */
+
+import bean from 'bean';
+import reportError from 'lib/report-error';
+import $ from 'lib/$';
+import config from 'lib/config';
+import detect from 'lib/detect';
+import fetchJson from 'lib/fetch-json';
+import mediator from 'lib/mediator';
+import userPrefs from 'common/modules/user-prefs';
+import { SearchTool } from 'facia/modules/onwards/search-tool';
+import type { CityPreference } from 'facia/modules/onwards/search-tool';
+
+let $holder = null;
+let searchTool = null;
+let eventsBound = false;
+const prefName = 'weather-location';
+
+type Location = {
+    id: string,
+    city: string,
+};
+
+type WeatherResponse = {
+    html: string,
+};
+
+const isNetworkFront = () =>
+    ['uk', 'us', 'au', 'international'].includes(config.page.pageId);
+
+export const Weather = {
+    init() {
+        if (!config.switches || !config.switches.weather || !isNetworkFront()) {
+            return false;
+        }
+
+        this.getDefaultLocation();
+    },
+
+    isNetworkFront() {
+        return ['uk', 'us', 'au', 'international'].includes(config.page.pageId);
+    },
+
+    /**
+     * Check if user has data in local storage.
+     * If yes return data from local storage else return default location data.
+     *
+     * @returns {object} geolocation - lat and long
```

Comment:
Could you please remove these doc (return-)type comments? There are not maintain and therefore don't have any value ... If you are motivated, it would be nice to see some Flow return types on the methods :)